### PR TITLE
Adding TOC on pages with more than 400 words

### LINKED ISSUE #2512: website (hugo): Add TOC on some pages larger than X words
AC:
 * Setup: https://gohugo.io/content-management/toc/#template-example-toc-partial

This might mean we need to move to new markdown engine.

### PR DESCRIPTION
**Pull request by soniasingla**
Signed-off-by: soniasingla <soniasingla.1812@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Fixes #2512 
This PR aims to add TOC to `getting-started.md` page

### CODE REVIEW

**Comment by soniasingla** [comment]
Yes, that would be best. I'm working on it :)

**Comment by soniasingla** [comment]
This pull request is setting up the automatic TOC above the header. I am blocked after this :)

File: website/layouts/_default/baseof.html [hunk 1]
```diff
@@ -26,6 +26,11 @@
 </head>
 
 <body>
+    {{ if or (eq .Params.toc true) (and (gt .WordCount 400) (ne .Params.toc false)) }}
+    <div class="column is-narrow">
+      {{ .TableOfContents }}
```
Comment (bwplotka) [inline website/layouts/_default/baseof.html:31]: Lol, that was easy =D

File: website/layouts/_default/baseof.html [hunk 2]
```diff
@@ -26,6 +26,11 @@
 </head>
 
 <body>
+    {{ if or (eq .Params.toc true) (and (gt .WordCount 400) (ne .Params.toc false)) }}
```
Comment (bwplotka) [inline website/layouts/_default/baseof.html:29]: :muscle:

**Comment by soniasingla** [comment]
@bwplotka before adding this code in HTML file, i tried by adding this code in a seperate new file but it doesn't reflect changes. In my opinion, the problem was we need to tell where to implement this TOC. I read the official documentation but can't find anything relevant to this :) 

> Ok, I think what we need to do is to NOT change HTML but actually all markdown pages. (:

File: website/layouts/_default/baseof.html (hunk 1 above)
Comment (soniasingla) [reply website/layouts/_default/baseof.html:31]: ❤️

**Comment by soniasingla** [comment]
Adding this code in `single.html` file (maybe)

```
{{ if or (eq .Params.toc true) (and (gt .WordCount 400) (ne .Params.toc false)) }} 
{{ .TableOfContents }}
{{ end }}
```
And, on the pages we want a TOC, setting `toc=true` might work.

+++
...
toc = true
...

I tried this method, but am having doubts with `single.html` file because the directory structure is bit confusing :) Using this method, provides us an advantage that our `HTML` structure is not disturbed :D

Reference : https://www.codeooze.com/webdesign/hugo-toc/

WDYT? @bwplotka @GiedriusS @povilasv 🙂

**Comment by bwplotka** [comment]
No no (: We need to put this in Markdown instead (: Because this is where we want to render it (: 

Now to automate it we can have some quick Perl to make it in scripts or we can put it manually to each markdown file in `docs` (:

**Comment by povilasv** [comment]
> {{ if or (eq .Params.toc true) (and (gt .WordCount 400) (ne .Params.toc false)) }} 
> {{ .TableOfContents }}
> {{ end }}

I like this approach :+1:

**Comment by povilasv** [comment]
In general I like this, but design seems a bit funky, maybe let's add a header and somehow make a list be less spaced out 

image

**Comment by soniasingla** [comment]
May be we can add something like `Working on Thanos` or `Thanos Documentation` or as header. WDYT? @povilasv :)

**Comment by povilasv** [comment]
I was thinking more like:

## Table Of Content
...

## Getting Started

**Comment by bwplotka** [comment]
I think what suggests POV makes sense, otherwise LGTM! :heart:

**Comment by bwplotka** [comment]
Amazing, let's merge it, thanks!

In separate PRs it would be awesome to fix few more things:
* Some links does not work: in case of header being link
* Some sides have manually created TOC, we should remove those :hugs: (https://deploy-preview-2689--thanos-io.netlify.app/contributing/coding-style-guide.md/)

Thanks :muscle:

**Pull request merged**